pkg/jobs/job_scheduler.go

```diff
@@ -360,8 +376,22 @@ func (s *jobScheduler) runDaemon(ctx context.Context, stopper *stop.Stopper) {
 			log.Errorf(ctx, "error registering executor metrics: %+v", err)
 		}
 
+		schedulerEnabledOnThisNode := func() bool {
+			if s.ShouldRunScheduler == nil || !schedulerRunsOnSingleNode.Get(&s.Settings.SV) {
+				return true
+			}
+
+			enabled, err := s.ShouldRunScheduler(ctx, s.DB.Clock().NowAsClockTimestamp())
+			if err != nil {
+				log.Errorf(ctx, "error determining if the scheduler enabled: %v; will recheck after %s",
+					err, recheckEnabledAfterPeriod)
+				return false
+			}
+			return enabled
+		}
+
 		for timer := time.NewTimer(initialDelay); ; timer.Reset(
-			getWaitPeriod(ctx, &s.Settings.SV, jitter, s.TestingKnobs)) {
+			getWaitPeriod(ctx, &s.Settings.SV, schedulerEnabledOnThisNode, jitter, s.TestingKnobs)) {
```

Comment:
Do we also want to call schedulerEnabledOnThisNode below after the timer triggers so that it is respected on the first iteration of the loop as well?

Reply:
Good point.